File: go/solutions/0456-132-pattern.go
package solutions

import "math"

func Find132pattern(nums []int) bool {
	st := []int{}
	third := math.MinInt32

	for i := len(nums) - 1; i >= 0; i-- {
		if nums[i] < third {
			return true
		}

		for len(st) > 0 && st[len(st)-1] < nums[i] {
			third = st[len(st)-1]
			st = st[:len(st)-1]
		}
		st = append(st, nums[i])
	}

	return false
}

File: js/0456-132-pattern.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var find132pattern = function (nums) {
  var st = [],
    third = Number.MIN_SAFE_INTEGER;
  for (var i = nums.length - 1; i >= 0; i--) {
    if (nums[i] < third) {
      return true;
    }

    while (st.length > 0 && st[st.length - 1] < nums[i]) {
      third = st[st.length - 1];
      st.pop();
    }

    st.push(nums[i]);
  }

  return false;
};

module.exports = { find132pattern };
